### ISSUE TITLE
[gatsby-transformer-remark] Filtering on tableOfContents in createResolvers makes Gatsby crash

### ISSUE THREAD

**Issue by problem233** [gatsbyjs/gatsby#17062]
## Description

Filtering on `tableOfContents` with `context.nodeModel.runQuery` in `createResolvers` makes Gatsby crash.

### Steps to reproduce

A minimal demo: https://github.com/problem233/gatsby-bug-demo

Related code: [link](https://github.com/problem233/gatsby-bug-demo/blob/0c2f0c3fd7c3ebda36c2eeadfd3cd3e32444186a/gatsby-node.js#L16)

Run `npm run dev`, open Graph*i*QL and run this query:

```graphql
query MyQuery {
  someType {
    toc
  }
}
```

### Expected result

Data:
```json
{
  "data": {
    "someType": {
      "toc": "<ul>\n<li>\n<p><a href=\"//#hello\">Hello</a></p>\n<ul>\n<li><a href=\"//#world\">World</a></li>\n</ul>\n</li>\n</ul>"
    }
  }
}
```

### Actual result

Gastby crashed.

### Environment

```
  System:
    OS: Linux 5.2 Arch Linux undefined
    CPU: (4) x64 Intel(R) Core(TM) i5-7300HQ CPU @ 2.50GHz
    Shell: 5.0.7 - /bin/bash
  Binaries:
    Node: 12.9.0 - /usr/bin/node
    npm: 6.11.2 - /usr/bin/npm
  Languages:
    Python: 3.7.4 - /usr/bin/python
  Browsers:
    Firefox: 68.0.2
  npmPackages:
    gatsby: ^2.13.79 => 2.13.82 
    gatsby-source-filesystem: ^2.1.16 => 2.1.18 
    gatsby-transformer-remark: ^2.6.19 => 2.6.19 
```

**Comment by stefanprobst** [comment]
There are a couple of issues here: the first is that the `tableOfContents` field currently requires a `pathToSlug` argument to produce a toc (not totally sure why tbh), and you cannot meaningfully run a filter on such a field. We actually plan to exclude fields that need an arg to produce a value from the filter type in the future. What you could do is get the field resolver from `info.schema.getType("MarkdownRemark").getFields().tableOfContents.resolve` and call it directly which is not pretty but should work.

The crash itself is because `tableOfContents` actually produces an invalid AST when no `pathToSlug` arg is provided which should be fixed in #17080

**Comment by problem233** [comment]
Sorry but I thought there was no way to get the resolver. Thanks!
